### ISSUE TITLE
GD-139: Plugin is broken after Update from 1.0.2 to 1.0.3

### ISSUE THREAD

**Issue by MikeSchulze** [MikeSchulze/gdUnit3#130]
**The used Godot version:**
Godot 3.3.2

**OS including version:**
all systems

**Describe the bug**
After update GdUnit3 form 1.0.2 to 1.0.3 the plugin is broken, only a Godot restart solves it

**Steps to Reproduce**

1. Install GdUnit3 form assetLib
2. activate it
3. the update popup is shown, press update
4. the update ends with success
5. but you see an error on the console indicating problems loading/updating classes

 ````
Godot Engine v3.3.2.stable.official (c) 2007-2021 Juan Linietsky, Ariel Manzur & Godot Contributors.
--- GDScript language server started ---
GdUnit3: Set inital settings 'gdunit3/settings/update_notification_enabled' to 'True'.
GdUnit3: Set inital settings 'gdunit3/settings/server_connection_timeout_minutes' to '30'.
GdUnit3: Set inital settings 'gdunit3/settings/test_timeout_seconds' to '300'.
GdUnit3: Set inital settings 'gdunit3/report/error_notification' to 'False'.
GdUnit3: Set inital settings 'gdunit3/report/verbose_orphans' to 'True'.
GdUnit3: Set inital settings 'gdunit3/report/assert/verbose_errors' to 'True'.
GdUnit3: Set inital settings 'gdunit3/report/assert/verbose_warnings' to 'True'.
GdUnit3: Server successfully started on port 31002
Loading GdUnit3 Plugin success
download immage: https://user-images.githubusercontent.com/347037/128760914-f91b2df6-3e5d-4dfb-ab0f-5e0d29eddb04.png
download immage: https://user-images.githubusercontent.com/347037/128761738-e6c3effa-8568-419f-9f5a-23b171752135.png
Update .. Downloading update ..
Update .. disable GdUnit3 ..
Closing all currently opened scripts ..
Unload GdUnit3 Plugin success
Update .. extracting zip 'C:/Users/Mike Schulze/AppData/Roaming/Godot/app_userdata/update-gdUnit-test/tmp/update/update.zip' to 'C:/Users/Mike Schulze/AppData/Roaming/Godot/app_userdata/update-gdUnit-test/tmp/update'
C:/Users/Mike Schulze/AppData/Roaming/Godot/app_userdata/update-gdUnit-test/tmp/update/update.zip successfully extracted
Update .. uninstall GdUnit3 ..
Update .. install new GdUnit3 version ..
Update .. refresh editor resources ..
Update .. executing patches ..
Update .. enable GdUnit3 ..
 res://addons/gdUnit3/src/ui/parts/InspectorToolBar.gd:16 - Invalid call. Nonexistent function 'init_version_label' in base 'GDScript'.
Update .. New GdUnit successfully installed
GdUnit3: Server successfully started on port 31002
Loading GdUnit3 Plugin success
````

Related to:
https://github.com/godotengine/godot/issues/29379
https://github.com/godotengine/godot/issues/51549

**Workaround**
Restart Godot after the update

**Comment by clemens-tolboom** [comment]
Not sure what version I got when starting Godot ... the update process is very nice ... but it ended into

```
Starting Godot using --editor
+ /Applications/Godot_v3.3.3-stable_osx.universal.app/Contents/MacOS/Godot --editor
arguments
0: /Applications/Godot_v3.3.3-stable_osx.universal.app/Contents/MacOS/Godot
1: --editor
Current path: /Users/clemens/Sites/Godot/fabrik
Godot Engine v3.3.3.stable.official.b973f997f - https://godotengine.org
OpenGL ES 3.0 Renderer: AMD Radeon R9 M370X OpenGL Engine
OpenGL ES Batching: ON
 
2021-09-06 15:46:25.865 Godot[35512:3528972] mac-virtualcam(DAL): PlugInMain version=1.3.0
2021-09-06 15:46:25.865 Godot[35512:3528972] mac-virtualcam(DAL): HardwarePlugIn_QueryInterface 
2021-09-06 15:46:25.866 Godot[35512:3528972] mac-virtualcam(DAL): HardwarePlugIn_Release sRefCount now = 0
2021-09-06 15:46:25.866 Godot[35512:3528972] mac-virtualcam(DAL): HardwarePlugIn_InitializeWithObjectID self=0x117321478
2021-09-06 15:46:25.866 Godot[35512:3528972] mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData OBSDALDevice(33) kCMIOObjectPropertyListenerAdded self=0x117321478 data(int)=1684629094
2021-09-06 15:46:25.866 Godot[35512:3528972] mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData OBSDALDevice(33) kCMIOObjectPropertyListenerAdded self=0x117321478 data(int)=1869180523
2021-09-06 15:46:25.866 Godot[35512:3528972] mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData OBSDALDevice(33) kCMIOObjectPropertyListenerAdded self=0x117321478 data(int)=1885762592
2021-09-06 15:46:25.950 Godot[35512:3528972] mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData OBSDALDevice(33) kCMIOObjectPropertyListenerAdded self=0x117321478 data(int)=1818850926
2021-09-06 15:46:25.950 Godot[35512:3528972] mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData OBSDALDevice(33) kCMIOObjectPropertyListenerAdded self=0x117321478 data(int)=1869180523
2021-09-06 15:46:25.950 Godot[35512:3528972] mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData OBSDALDevice(33) kCMIOObjectPropertyListenerAdded self=0x117321478 data(int)=1937009955
2021-09-06 15:46:25.950 Godot[35512:3528972] mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData OBSDALDevice(33) kCMIOObjectPropertyListenerAdded self=0x117321478 data(int)=1886156132
2021-09-06 15:46:25.950 Godot[35512:3528972] mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData OBSDALDevice(33) kCMIOObjectPropertyListenerAdded self=0x117321478 data(int)=1935833461
2021-09-06 15:46:25.951 Godot[35512:3528972] mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData OBSDALStream(34) kCMIOObjectPropertyListenerAdded self=0x117321478 data(int)=1885762592
Registered camera OBS Virtual Camera with id 1 position 0 at index 0
Registered camera FaceTime HD Camera (Display) with id 2 position 0 at index 1
Registered camera FaceTime HD Camera with id 3 position 0 at index 2
GdUnit3: Set inital settings 'gdunit3/settings/update_notification_enabled' to 'True'.
GdUnit3: Set inital settings 'gdunit3/settings/server_connection_timeout_minutes' to '30'.
GdUnit3: Set inital settings 'gdunit3/settings/test_timeout_seconds' to '300'.
GdUnit3: Set inital settings 'gdunit3/report/error_notification' to 'False'.
GdUnit3: Set inital settings 'gdunit3/report/verbose_orphans' to 'True'.
GdUnit3: Set inital settings 'gdunit3/report/assert/verbose_errors' to 'True'.
GdUnit3: Set inital settings 'gdunit3/report/assert/verbose_warnings' to 'True'.
GdUnit3: Server successfully started on port 31002
Loading GdUnit3 Plugin success
download immage: https://user-images.githubusercontent.com/347037/132136825-68fcbe72-a3d7-4751-8a4e-0f36a7234c7e.png
Update .. Downloading update ..
Update .. disable GdUnit3 ..
Closing all currently opened scripts ..
Unload GdUnit3 Plugin success
Update .. extracting zip '/Users/clemens/Library/Application Support/Godot/app_userdata/fabrik/tmp/update/update.zip' to '/Users/clemens/Library/Application Support/Godot/app_userdata/fabrik/tmp/update'
Detect OS : OSX
/Users/clemens/Library/Application Support/Godot/app_userdata/fabrik/tmp/update/update.zip successfully extracted
Update .. uninstall GdUnit3 ..
Update .. install new GdUnit3 version ..
Update .. refresh editor resources ..
ERROR: _load: Resource file not found: res://addons/gdUnit3/src/update/132136825-68fcbe72-a3d7-4751-8a4e-0f36a7234c7e.png.
   At: core/io/resource_loader.cpp:287.

======= LOTS MORE Resource ERRORs ========

ERROR: _load: Resource file not found: res://addons/gdUnit3/src/update/132136825-68fcbe72-a3d7-4751-8a4e-0f36a7234c7e.png.
   At: core/io/resource_loader.cpp:287.
Update .. executing patches ..
Update .. enable GdUnit3 ..
Update .. New GdUnit successfully installed
GdUnit3: Server successfully started on port 31002
Loading GdUnit3 Plugin success
ERROR: remove_child: Cannot remove child node Control2 as it is not a child of this node.
   At: scene/main/node.cpp:1242.
ERROR: remove_child: Parent node is busy setting up children, remove_node() failed. Consider using call_deferred("remove_child", child) instead.
   At: scene/main/node.cpp:1219.
ERROR: ~Node: Condition "data.parent" is true.
   At: scene/main/node.cpp:3036.
Unload GdUnit3 Plugin success
handle_crash: Program crashed with signal 11
```

On the terminal I had to `CTRL-C` after that last line

**Comment by MikeSchulze** [comment]
The update notify says "After upgrading a restart of Godot is strongly recommended"
The update fixes the "remove_child: Cannot remove child node Control2 as it is not a child of this node"
This error was related to a wrong added node and remove where is fixed in 1.0.4

**Comment by MikeSchulze** [comment]
You can test the fix by simple enable/disable the plugin, the plugin should load and unload without errors.

**Comment by MikeSchulze** [comment]
fixed since 1.0.4
